### PR TITLE
Make List.Rest.iterator() O(1)

### PR DESCRIPTION
**Pull request by sadmac7000**

### CODE REVIEW

File: src/ceylon/language/List.ceylon [hunk 1]
```diff
@@ -851,13 +851,12 @@ shared interface List<out Element=Anything>
...
         iterator() 
-                => let (iter = outer.iterator()) 
+                => let (o = outer)
             object satisfies Iterator<Element> {
                 variable value i=0;
-                while (i++<from) {
-                    iter.next();
-                }
-                next() => iter.next();
+                next() => if (i < outer.size)
```
Comment (gavinking) [inline src/ceylon/language/List.ceylon]: @sadmac7000 that doesn't look quite right. Shouldn't it be `if (i < outer.size-from)`, or, even better, just start at `value i = from`?

**Comment by sadmac7000** [comment]
`outer` is the `Rest` object itself. `outer.outer` is the original list.

**Comment by gavinking** [comment]
OIC.

**Comment by gavinking** [comment]
Thanks.

**Pull request merged**